test/Table_Tests/src/Common_Table_Operations/Core_Spec.enso

```diff
@@ -80,6 +82,32 @@ spec setup =
             table.get (Pair.new 1 2) . should_fail_with Illegal_Argument.Error
             table.get (Pair.new 1 2) . catch . to_display_text . should_equal "Illegal Argument: expected 'selector' to be either a Text or an Integer, but got Pair.Value."
 
+    Test.group prefix+"Table.set" <|
+        Test.specify "should allow adding a column" <|
+            bar2 = table.get "bar" . rename "bar2"
+            t2 = table.set bar2
+            t2.column_names . should_equal ["foo", "bar", "Baz", "foo_1", "foo_2", "ab.+123", "abcd123", "bar2"]
+            t2.get "bar2" . to_vector . should_equal [4, 5, 6]
+
+            t3 = t2.set bar2 "bar3"
+            t3.column_names . should_equal ["foo", "bar", "Baz", "foo_1", "foo_2", "ab.+123", "abcd123", "bar2", "bar3"]
+
+        Test.specify "should allow replacing a column" <|
+            foo = table.get "bar" . rename "foo"
+            t2 = table.set foo
+            t2.column_names . should_equal ["foo", "bar", "Baz", "foo_1", "foo_2", "ab.+123", "abcd123"]
+            t2.get "foo" . to_vector . should_equal [4, 5, 6]
+
+            t3 = t2.set foo "bar3"
+            t3.column_names . should_equal ["foo", "bar", "Baz", "foo_1", "foo_2", "ab.+123", "abcd123", "bar3"]
+
+        Test.specify "should allow adding a column" <|
+            bar2 = table.get "bar" . rename "bar2"
+            table.set bar2 set_mode=Set_Mode.Update . should_fail_with Missing_Column.Error
+
+            foo = table.get "bar" . rename "foo"
+            table.set foo set_mode=Set_Mode.Add . should_fail_with Existing_Column.Error
```

Comment:
For a while already, when doing stuff with changing columns I was thinking if it may be worth adding the following test scenario:

```
Test.specify "should not affect existing columns that depended on the old column being replaced" <|
    t1 = table_builder [["X", [1,2,3]]]
    t2 = t1.set (t1.at "X" * 100) new_name="Y"
    t3 = t2.set "[X] + 10" new_name="Z"
    t4 = t3.set "[X] + 1000" new_name="X"
    
    t4.at "X" . to_vector . should_equal [1001, 1002, 1003]
    t4.at "Y" . to_vector . should_equal [100, 200, 300]
    t4.at "Z" . to_vector . should_equal [11, 12, 13]
```

I.e. we check that when I replace a column with a new one, any other columns that depended on that one are not changed, because they referred to the old one and not the new one, and the reference is correctly preserved.

While I'm pretty sure our implementation is safe from this issue, I can easily imagine an implementation doing this wrong (especially one like our SQL generator). So it may be worth just to: double-check ours works + ensure no regressions.